### PR TITLE
Do not apply evacuation budgets in non-generational mode

### PR DESCRIPTION
**Pull request by earthling-amzn**
CSet selection for non-generational mode was using a value configured on the young generation. In non-generational modes we do not maintain all the attributes of the young generation (instead we maintain the _global_ generation). This caused many cycles in which the maximum CSet was zero (or close to it). This, in turn, caused the collector to run much more frequently (approximately 3x on specjbb) which caused severe performance regression in critical jops.

I'm not sure why the diff algorithm is struggling so much with these changes. I pulled up the `mode()->is_generational()` out of `compute_evacuation_budgets` and `adjust_evacuation_budgets` into the caller and replaced the check with an assert (re-indenting the code in the method).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah pull/171/head:pull/171` \
`$ git checkout pull/171`

Update a local copy of the PR: \
`$ git checkout pull/171` \
`$ git pull https://git.openjdk.org/shenandoah pull/171/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 171`

View PR using the GUI difftool: \
`$ git pr show -t 171`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/171.diff">https://git.openjdk.org/shenandoah/pull/171.diff</a>

</details>

### CODE REVIEW

**Review by kdnilsen — COMMENTED** [review]
Thanks for this fix.  I think I may have introduced the bad code.

**Comment by ysramakrishna** [comment]
> I'm not sure why the diff algorithm is struggling so much with these changes. 

Are the diff struggles in git (I assume), or in github (I assume not). I have seen git have issues with indentation that caused issues with merges in like manner recently. :-(

I wonder if there's a git option for indentation that would rended this diff better?

> ```
> --indent-heuristic
> Enable the heuristic that shifts diff hunk boundaries to make patches easier to read. This is the default.
> 
> --no-indent-heuristic
> Disable the indent heuristic.
> ```

**Comment by earthling-amzn** [comment]
The diff looked hairy even in `p4merge`.

**Comment by ysramakrishna** [comment]
Interesting, I wonder if this would make any diff (pun not intended):

```
--diff-algorithm={patience|minimal|histogram|myers}
Choose a diff algorithm. The variants are as follows:

default, myers
The basic greedy diff algorithm. Currently, this is the default.

minimal
Spend extra time to make sure the smallest possible diff is produced.

patience
Use "patience diff" algorithm when generating patches.

histogram
This algorithm extends the patience algorithm to "support low-occurrence common elements".

For instance, if you configured the diff.algorithm variable to a non-default value and want to use the default one, then you have to use --diff-algorithm=default option.
```

**Comment by earthling-amzn** [comment]
I think my editor applied some bonus formatting... One second, I will fix up the patch.

**Comment by earthling-amzn** [comment]
Try hiding whitespace on github - looks much better to me:

image

File: src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp [hunk 1]
```diff
@@ -791,40 +779,40 @@ void ShenandoahGeneration::prepare_regions_and_collection_set(bool concurrent) {
       ShenandoahFinalMarkUpdateRegionStateClosure old_cl(nullptr);
       heap->old_generation()->parallel_heap_region_iterate(&old_cl);
     }
-
-    heap->assert_pinned_region_status();
```
Comment (earthling-amzn) [inline src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp:795]: This is called again immediately inside collection set selection so I removed this invocation.

**Review by kdnilsen — APPROVED** [review]

**Pull request closed without merging**